### PR TITLE
Add Slack Events API adapter for Express

### PR DESCRIPTION
**Pull request by ktorng**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,2 +1,94 @@
+// Load environment variables
```
Comment (voxlol) [inline index.js]: rm comment

File: index.js [hunk 2]
```diff
@@ -1,2 +1,94 @@
...
 const express = require('express');
-const slack = require('slack');
\ No newline at end of file
+const bodyParser = require('body-parser');
+
+const { getClientByTeamId } = require('./util/getClient');
+
+const port = process.env.APP_PORT || 3000;
+
+// Initialize a data structures to store team authorization info (typically stored in a database)
```
Comment (voxlol) [inline index.js]: rm comment imo, well written code wont need comments (lets introduce comments once our app gets complex, not when its <100 lines of code)

File: index.js [hunk 3]
```diff
@@ -1,2 +1,94 @@
...
+const botAuthorizations = {}
+
+// *** Initialize event adapter using verification token from environment variables ***
+const slackEvents = slackEventsApi.createSlackEventAdapter(
```
Comment (voxlol) [inline index.js]: style change, but how do you feel about making this all 1 line?

File: index.js [hunk 4]
```diff
@@ -1,2 +1,94 @@
...
+  process.env.SLACK_VERIFICATION_TOKEN, 
+  { includeBody: true }
+);
+
+// Initialize Add to Slack (OAuth) helpers
+passport.use(new SlackStrategy({
```
Comment (voxlol) [inline index.js]: Just a side note - lets move this all to an authentication.js file in another pR

File: index.js [hunk 5]
```diff
@@ -1,2 +1,94 @@
...
+  done(null, {});
+}));
+
+// Initialize an Express application
+const app = express();
+app.use(bodyParser.json());
+
+// Plug the Add to Slack (OAuth) helpers into the express app
+app.use(passport.initialize());
+app.get('/', (req, res) => {
```
Comment (voxlol) [inline index.js]: Is this endpoint necessary? Can we get rid of it?

File: index.js [hunk 6]
```diff
@@ -1,2 +1,94 @@
...
+  res.send('<a href="/auth/slack"><img alt="Add to Slack" height="40" width="139" src="https://platform.slack-edge.com/img/add_to_slack.png" srcset="https://platform.slack-edge.com/img/add_to_slack.png 1x, https://platform.slack-edge.com/img/add_to_slack@2x.png 2x" /></a>');
+});
+app.get('/auth/slack', passport.authenticate('slack', {
+  scope: ['bot']
```
Comment (voxlol) [inline index.js]: When a hash only has 1 or 2 keys, can we make it 1-line? Thoughts?

File: index.js [hunk 7]
```diff
@@ -1,2 +1,94 @@
...
+
+// *** Plug the event adapter into the express app as middleware ***
+app.use('/slack/events', slackEvents.expressMiddleware());
+
+// *** Attach listeners to the event adapter ***
+
+// *** Greeting any user that says "hi" ***
+slackEvents.on('message', (message, body) => {
+  // Only deal with messages that have no subtype (plain messages) and contain 'hi'
+  if (!message.subtype && message.text.indexOf('hi') >= 0) {
```
Comment (voxlol) [inline index.js]: Can we make this check more strict by requiring the message to be exclusive "hi"? We are getting too many dev messages currently

File: index.js (hunk 4 above)
Comment (voxlol) [inline index.js]: 👍 nice passport

**Review by voxlol — CHANGES_REQUESTED** [review]
Good work, but can you remove all the comments? I don't think comments are necessary if our code is well-written. Thoughts?

**Comment by ktorng** [comment]
agreed, cleaned

**Review by voxlol — APPROVED** [review]

**Pull request merged**